### PR TITLE
Add documentation for 'format' table function

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add documentation for 'format' table function

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by davenger** [comment]
Shouldn't this file name be 'format.md' with '.' instead of ','
[docs/zh/sql-reference/table-functions/format,md](https://github.com/ClickHouse/ClickHouse/pull/42779/files#diff-aae72141e3565d76d9065a552f3e0d6ab8fa6c187aa344ad2606fd249a887eba)

**Comment by Avogar** [comment]
> Shouldn't this file name be 'format.md' with '.' instead of ','?

That's a typo, thanks for noticing

File: docs/en/sql-reference/table-functions/format.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+**Syntax**
+
+``` sql
+format(format_name, data)
+```
+
+**Parameters**
+
+-   `format_name` — The [format](../../interfaces/formats.md#formats) of the data.
+-   `data` — String literal containing data in specified format
```
Comment (qoega) [inline docs/en/sql-reference/table-functions/format.md:20]: Do we support column as well as input?

Comment (alexey-milovidov) [reply docs/en/sql-reference/table-functions/format.md:20]: Table functions cannot get columns from other tables.
Maybe you meant constant expressions?

Looks like yes:

```
milovidov-desktop :) SELECT * FROM format(LineAsString, 'Hello' || 'World')

┌─line───────┐
│ HelloWorld │
└────────────┘

1 row in set. Elapsed: 0.001 sec.
```

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Maybe it's a burden and we'd better do it once with a script, but... let's also copy the docs into the Documenation object in Factory.

**Pull request merged**